src/animation/play_once_animation.rb

```diff
@@ -0,0 +1,22 @@
+require 'gosu'
+require 'src/animation'
+
+class PlayOnceAnimation < Animation
+  def initialize(*args)
+    super(*args)
+    @played_frames = []
+  end
+
+  def start
+    frame = super
+    @played_frames << frame
+    frame
+  end
+
+  def stop
+    return start unless (frames - @played_frames).empty?
```

Comment:
Yeah, I tried that, it gets buggier. Thing is `Gosu.milliseconds` returns `0` at the first call, and then it is unpredictable, it keeps counting. This can lead to skipping a frame, so sometimes the `@played_frames.size >= frames.size` even though the animation has not completed. I need to search for a more reliable way to do animations.